hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+By contrast, the largest possible `generate` command allowed by [NIST SP 800-90A](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf) is for 2<sup>19</sup> bits (or 4096 AES codewords).
+Assuming an AES encryption delay of 16 clocks, and a 100 MHz clock frequency, the longest allowable `generate` command would take only 0.7 ms to complete.
+
+### Note on Sharing of CSRNG Instance State Variables
+
+Once an application interface port has received an `instantiate` command, that application interface port then has access to a unique CSRNG instance, which is shared by all endpoints on the same EDN.
+Therefore from a security perspective, an attack to that particular CSRNG instance is an attack on all the endpoints that share the same EDN.
+Meanwhile, seeds and other state variables specific to a particular CSRNG instance are not shared between endpoints on *separate* EDN instances, or with any hardware devices with direct connections to dedicated CSRNG application interface ports.
+
+## Interactions with Peripheral Devices
+
+Peripheral ports distribute data to the endpoints using four signals: `req`, `ack`, `bus`, and `fips`.
+
+The EDN is activated in request/acknowledge mode whenever both `EDN_ENABLE` and `BCAST_MODE_DIS` bits are simultaneously asserted in {{< regref "CTRL" >}}.
+In request/acknowledge mode, fresh (i.e. previously unseen) random values are distributed to the endpoints via the parametrizable-width `bus` signal, in response to an `req` signal.
+Whenever new values are placed on the `bus`, the `ack` is asserted until the values are consumed by the endpoint, as indicated by simultaneous assertion of the `req` and `ack` signals in the same cycle.
+Otherwise `ack` is deasserted until enough fresh bits are received from CSRNG.
+
+The `fips` signal is used to identify whether the values received on the `bus`, have prepared with complete adherence to the recommendations in NIST SP 800-90.
+Typically if the `fips` signal is deasserted, it means the associated CSRNG instance has been instantiated with a pre-FIPS seed.
+
+Endpoints can exercise careful control of the `req` and `ack` signals to manage how long random values stay asserted on the bus.
+For instance, if fresh random numbers are asserted on `bus`, but the endpoint deasserts `req` in the same cycle, those random values are not consumed and will be left unchanged until the endpoint asserts `req` to consume them.
+The `req` signal can be asserted even when fresh values are not required, to simply clear the bus of already used data, or to eliminate potentially stale data, meaning data which may have been sitting on the bus longer than security considerations would deem acceptable.
+When using the `req` signal to clear already consumed or stale data from the `bus`, the `ack` signal is monitored to confirm that the undesired data has been successfully discarded.
```

Comment:
Could we add a timing diagram to illustrate this? I'm a little confused, because I interpreted the req/ack interface as being something like:
1. Device asserts `req` because it wants some random data
1. Some time passes.
1. EDN asserts `ack` and puts data on the bus.
1. Device drops `req`.
1. EDN clears bus.

Do we imagine that devices are going to "change their minds" about requesting data? If not, we could disallow it, which I think would solve this problem. Alternatively, the design could ensure that "ack" depends on "req" (through a flop, giving a single-cycle delay) and never leave data on the bus.

Even if we don't decide to change this, we could still clear `edn_bus` when `edn_ack` is low. Looking at the broadcast timing diagram below, I think that would probably make the data on that bus rather harder to snoop.

Reply:
Not change their minds per se, but is is common for devices to sometimes have an internal stall condition for these type of buses, and temporarily deassert Req.  (However some protocols like AXI stream prohibit this).

No one asked for this paragraph.  I'll just remove it entirely.